assayist/processor/base.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from abc import ABC, abstractmethod
+import json
+import neomodel
+import os
+
+from assayist.common.models import content, source
+from assayist.processor.configuration import config
+from assayist.processor.logging import log
+
+
+class Analyzer(ABC):
+    """Base Abstract class that analyzers will inherit from."""
+
+    METADATA_DIR = '/metadata/'
+    MESSAGE_FILE = METADATA_DIR + 'message.json'
+    BUILD_FILE = METADATA_DIR + 'buildinfo.json'
+    TASK_FILE = METADATA_DIR + 'taskinfo.json'
+    RPM_FILE = METADATA_DIR + 'rpms.json'
+    ARCHIVE_FILE = METADATA_DIR + 'archives.json'
+    IMAGE_RPM_FILE = METADATA_DIR + 'image-rpms.json'
+    BUILDROOT_FILE = METADATA_DIR + 'buildroot-components.json'
+
+    def main(self):
+        """Call this to run the analyzer."""
+        neomodel.config.DATABASE_URL = config.DATABASE_URL
+        neomodel.config.AUTO_INSTALL_LABELS = True
+        # run the analyzer in a transaction
+        neomodel.db.begin()
+        try:
+            self.run()
+            log.debug("Analyzer completed successfully, committing.")
+            neomodel.db.commit()
+        except Exception as e:
+            log.exception("Error encountered executing Analyzer, rolling back transaction.")
+            neomodel.db.rollback()
+            raise
+
+    @abstractmethod
+    def run(self):
+        """Implement anlyzer code here in your subclass."""
+
+    def read_metadata_file(self, FILE):
```

Comment:
Optional: How about making this a private method? I don't this should be called outside of the class.

Reply:
I don't think I'm going to bother changing this. It may not ever happen, but something could theoretically decide to use this method to read a file even if it's not a class / subclass. Also I'm not sure what implications the name mangling of a private method would be for mocking out the method in the tests, and I don't care enough to find out.